### PR TITLE
Finish up argument parsing

### PR DESCRIPTION
**Pull request by jaygel179**
changes
-----------
* added arg-parser

usage
--------
* `$ node index.js path/to/component.html`
* `$ node index.js -t css path/to/component.html`

sample output
-------------------
```
$ node index.js -h
Splitter for ractive.js component files.
usage: index [options] <path>

 <path>            path to file

 -h, --help        display help & usage
 -t, --type=TYPE   output file type, default [script], types: [css|script|modules|imports|temp]
 -v, --version     display cli name & version

 https://github.com/engagespark/racsplit
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,9 +1,15 @@
-#!/usr/bin/node
+#!/usr/bin/env node
```
Comment (mknecht) [inline index.js]: :+1:

File: index.js [hunk 2]
```diff
@@ -26,20 +32,20 @@ var TYPE_EXTRACTER = {
...
+var type = argv.type;
+var filePath = argv.path;
 
 if (VALID_TYPES.indexOf(type) < 0) {
    console.error('Invalid type. Valid types are: ' + VALID_TYPES.join(', '));
    process.exit();
 }
 
-if (!fileName) {
+if (!filePath) {
```
Comment (mknecht) [inline index.js]: good name change!

File: arguments.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            "full" : "type",
+            "short" : "t",
+            "value" : true,
+            "description" : "'css|script|modules|imports|template"
+        },
+        {
+            "full" : "path",
+            "short" : "p",
+            "value" : true,
+            "description" : "File path"
```
Comment (mknecht) [inline arguments.json]: the one thing that's definitely mandatory is the path, right? How about making it not an option, but a positional argument? i.e.

```racsplit -t css path/to/module.html```

File: arguments.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        }
+    ],
+    "defaults" : {
+        "help" : true,
+        "helpExtras" : "Documentation can be fount at github.com/engagespark/racsplit/blob/master/README.md"
```
Comment (mknecht) [inline arguments.json]: * found
* put protocol, so console will auto-link it, and just reference the project, as the readme will be shown! :) https://github.com/engagespark/racsplit

File: index.js [hunk 5]
```diff
@@ -1,9 +1,15 @@
...
 var sander = require('sander');
+var argumentParser = require("node-argument-parser");
+var argv = argumentParser.parse("./arguments.json", process);
 
+if (argv.help) {
+    process.exit();
+}
 
 // first arg goes to index 2 process.arg[2]
 // types [css|script|modules|imports|template]
```
Comment (mknecht) [inline index.js]: remove comment, as it's outdated and not necessary anymore.

File: index.js [hunk 6]
```diff
@@ -26,20 +32,20 @@ var TYPE_EXTRACTER = {
...
 
-var type = process.argv[2];
-var fileName = process.argv[3];
+var type = argv.type;
+var filePath = argv.path;
 
 if (VALID_TYPES.indexOf(type) < 0) {
    console.error('Invalid type. Valid types are: ' + VALID_TYPES.join(', '));
    process.exit();
 }
```
Comment (mknecht) [inline index.js]: i don't think you need this anymore, with the argparser doing the validation.

Comment (jaygel179) [inline index.js]: argparser doesn't validate the inputs

**Pull request merged**